### PR TITLE
RASS-314 Add conviction date field to add/edit sentence

### PR DESCRIPTION
**Pull request by pavilionsahota**

### CODE REVIEW

File: server/views/pages/offence/edit-offence.njk [hunk 1]
```diff
@@ -59,6 +59,20 @@
                         }]
                     }
                 }].concat(summaryRows,[{
+                    key: {
+                        text: "Conviction date"
+                    }, value: {
+                        text: (offence.sentence.convictionDate | formatDate if offence.sentence and offence.sentence.convictionDate else 'N/A')
```
Comment (carloveo-moj) [inline server/views/pages/offence/edit-offence.njk:65]: you're already in an if statement here https://github.com/ministryofjustice/hmpps-remand-and-sentencing/pull/423/files#diff-4624e738415f013f715468bbeda39bca7ba921f3cd278bad12306572a0757897L46 so ` offence.sentence` condition is always true, can just drop this if statement down to `if offence.sentence.convictionDate`

File: server/routes/offenceRoutes.ts [hunk 2]
```diff
@@ -1030,8 +1031,19 @@ export default class OffenceRoutes {
       addOrEditCourtCase,
       addOrEditCourtAppearance,
     } = req.params
+    const offence = this.getSessionOffenceOrAppearanceOffence(req, nomsId, courtCaseReference, offenceReference)
```
[the remark this answers is not in the document]
Comment (pavilionsahota) [reply server/routes/offenceRoutes.ts:1034]: I have adjusted this code in the way you have described

File: server/views/pages/offence/edit-offence.njk (hunk 1 above)
Comment (pavilionsahota) [reply server/views/pages/offence/edit-offence.njk:65]: I have adjusted this code in the way you have described

**Review by carloveo-moj — APPROVED** [review]
LGTM

File: server/routes/offenceRoutes.ts [hunk 3]
```diff
@@ -1030,12 +1031,17 @@ export default class OffenceRoutes {
       addOrEditCourtCase,
       addOrEditCourtAppearance,
     } = req.params
+    let offenceConvictionDateForm: OffenceConvictionDateForm
```
Comment (carloveo-moj) [inline server/routes/offenceRoutes.ts:1034]:
```suggestion
    const offenceConvictionDateForm = (req.flash('offenceConvictionDateForm')[0] || {}) as OffenceConvictionDateForm
```

need the just submitted form to show values correctly when displaying errors

**Review by carloveo-moj — APPROVED** [review]
LGTM

**Pull request merged**